src/main/resources/assets/js/components/AppListComponent.jsx

```diff
@@ -12,17 +12,13 @@ var AppListComponent = React.createClass({
 
   propTypes: {
     collection: React.PropTypes.object.isRequired,
-    onSelectApp: React.PropTypes.func.isRequired
+    router: React.PropTypes.object.isRequired
```

Comment:
Passing a router all the way down the tree is the same thing as creating a global. It suggests that we've not encapsulated things correctly and need to come at it from a different perspective.